### PR TITLE
Improve beacon chain coverage Part 1

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Other

related to #10047 
**What does this PR do? Why is it needed?**

As we get closer to the merge we would like much more test coverage for our beacon node.

- [] ~~removed enable native internal feature boolean and reversed all the disable native state conditions to represent the same thing.
this cleans up the code and defaults the state to native state for unit tests.~~
rolling back due to instability using both native state and proto state in unit tests, they are starting to affect each other due to how it is cloned/copied...
- [x] improve fieldtrie package coverage (~50 to 82.5% average)
- [x] improve beacon chain node package coverage (~58 to 70% average)
- [x] improve state-native package coverage (~56.1 to 69.4% average)

future PRs will be opened for more coverage

### CODE REVIEW

File: beacon-chain/node/node_test.go [hunk 1]
```diff
@@ -31,15 +44,90 @@ func TestNodeClose_OK(t *testing.T) {
...
+	go func() {
+		node.Start()
+	}()
+	time.Sleep(3 * time.Second)
+	node.Close()
+	require.LogsContain(t, hook, "Starting beacon node")
+
+}
+
+func TestNodeStart_Ok_registerDeterminsticGenesisService(t *testing.T) {
```
Comment (rkapka) [inline beacon-chain/node/node_test.go:85]:
```suggestion
func TestNodeStart_Ok_registerDeterministicGenesisService(t *testing.T) {
```

File: beacon-chain/state/state-native/proofs_test.go [hunk 2]
```diff
@@ -4,61 +4,100 @@ import (
...
+		require.ErrorContains(t, "not supported", err)
+	})
+	t.Run("finalized root", func(t *testing.T) {
+		_, err := st.FinalizedRootProof(ctx)
+		require.ErrorContains(t, "not supported", err)
+	})
+	features.Init(&features.Flags{EnableNativeState: false})
+}
+
+func TestBeaconStateMerkleProofs_bellatrix(t *testing.T) {
```
Comment (rkapka) [inline beacon-chain/state/state-native/proofs_test.go:34]: Can you also write one for Altair?

File: validator/rpc/auth_token_test.go [hunk 3]
```diff
@@ -50,6 +50,7 @@ func TestServer_AuthenticateUsingExistingToken(t *testing.T) {
 	require.Equal(t, true, len(srv.jwtSecret) > 0)
 	_, err = srv.JWTInterceptor()(ctx, "xyz", unaryInfo, unaryHandler)
 	require.NoError(t, err)
+
```
Comment (rkapka) [inline validator/rpc/auth_token_test.go:53]:
```suggestion
```

**Comment by rkapka** [comment]
You need to resolve conflicts

File: beacon-chain/state/state-native/proofs_test.go (hunk 2 above)
Comment (james-prysm) [reply beacon-chain/state/state-native/proofs_test.go:104]: added something similar

**Review by rkapka — APPROVED** [review]

**Pull request merged**